kafka-protocol-binding.md

```diff
@@ -130,14 +142,19 @@ this specification then it is not a valid CloudEvent.
 If the `content-type` header is not present then the receiver uses
 _structured_ mode with the JSON event format.
 
-### 3.1. Key Attribute
+### 3.1. Key mapping
+
+The 'key' of the Kafka message is populated by a "Key Mapper" function, which
+might map the key directly from one of the CloudEvent's attributes, but might
+also use information from the application environment, from the CloudEvent's
+data or other sources.  
+
+The shape and configuration of the "Key Mapper" function is implementation
+specific.
 
-The 'key' attribute is populated by a partitionKeyExtractor function. The
-partitionKeyExtractor is a protocol specific function that contains bespoke
-logic to extract and populate the value. The key attribute MUST be encoded
-as UTF-8 string both in the Kafka message and in the CloudEvents extension.
-A default implementation of the extractor will use the
-[Partitioning](extensions/partitioning.md) extension value.
+Every implementation SHOULD provide a default "Key Mapper" implementation that
+maps the [Partitioning](extensions/partitioning.md) `partitionkey` attribute
+value to the 'key' of the Kafka message as-is, if present.
```

Comment:
Other than writing the value of `partitionKey` as `key` of kafka message, should the implementation write the `partitionKey` as extension too? Or it applies the same rule of `content-type` (aka i write content-type but not `ce_datacontenttype`) ?

Reply:
If the value is part of the event, it must also be written into the event. A different key mapper function could take OOB information and use that for the key. I'll add a clarification.

Reply:
Okay so the event, in that case, is left unchanged.